gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/language/translator/GroovyTranslateVisitor.java

```diff
@@ -113,4 +118,63 @@ public Void visitFloatLiteral(final GremlinParser.FloatLiteralContext ctx) {
         }
         return null;
     }
+
+    @Override
+    public Void visitInfLiteral(final GremlinParser.InfLiteralContext ctx) {
+        if (ctx.SignedInfLiteral().getText().equals("-Infinity")) {
+            sb.append("Double.NEGATIVE_INFINITY");
+        } else {
+            sb.append("Double.POSITIVE_INFINITY");
+        }
+        return null;
+    }
+
+    @Override
+    public Void visitNullLiteral(final GremlinParser.NullLiteralContext ctx) {
+        if (ctx.getParent() instanceof GremlinParser.GenericLiteralMapNullableArgumentContext) {
+            sb.append("null as Map");
+            return null;
+        }
+
+        sb.append(ctx.getText());
+        return null;
+    }
+
+    @Override
+    public Void visitTraversalSourceSpawnMethod_inject(final GremlinParser.TraversalSourceSpawnMethod_injectContext ctx) {
+        return handleInject(ctx);
+    }
+
+    @Override
+    public Void visitTraversalMethod_inject(final GremlinParser.TraversalMethod_injectContext ctx) {
+        return handleInject(ctx);
+    }
+
+    private Void handleInject(final ParserRuleContext ctx) {
```

Comment:
please expand the comment a bit to include "why" it can't work correctly - `inject()` ends up being ambiguous with groovy's jdk extension of `inject(Object initialValue, Closure closure)`.  not sure why `inject(Closure)` doesn't seem to conflict with inject(null)` - a groovy mystery? i wonder if that is just coincidental? should we just account for it to avoid future issues?

i think it would be good to add a specific tests for these cases in the `GremlinTranslatorTest`.